File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.js
/**
 * @param {string} num
 * @return {string}
 */
var largestOddNumber = function(num) {
    let pR = num.length - 1;
    while (pR >= 0) {
        if (Number(num[pR]) % 2 === 0) pR--;
        else break;
    }
    return pR >= 0 ? num.substring(0, pR + 1) : "";
};

File: 1903-largest-odd-number-in-string/1903-largest-odd-number-in-string.kt
class Solution {
    fun largestOddNumber(num: String): String {

        val chunked = num.chunked(1)

        for (index in chunked.lastIndex downTo 0) {
            if (chunked[index].toInt() % 2 == 0 || chunked[index].toInt() == 0) {
                continue
            } else {
                return num.substring(0, index + 1)
            }
        }

        return ""
    }
}